File: civitasJava/src/civitas/TituloExamen.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

/**
 *
 * @author Esther
 */
public class TituloExamen extends TituloPropiedad{
    
    TituloExamen(String nombre, float ab, float fr, float hb, float pc, float pe) {
        super(nombre, ab, fr, hb, pc, pe);
    }
    
    @Override
    protected boolean construirCasa(Jugador jugador){
        boolean ret = super.construirCasa(jugador);
        jugador.paga(precioEdificar);
        
        return ret;
    }
}

File: civitasRuby/lib/civitas/titulo_examen.rb
#encoding:utf-8

=begin
Authors: Esther García Gallego
         Yesenia González Dávila
         Grupo B3
=end

module Civitas

class TituloExamen < TituloPropiedad
  
  def initialize 
    
  end
  
  def initialize (nombre, ab, fr, hb, pc, pe)
    super(nombre, ab, fr, hb, pc, pe)
  end
  
  def construirCasa(jugador)
    result = super(jugador)
    if result # SUmar otra vez el dinero que ha costado al sañdo
      jugador.paga(@precioEdificar)
    end
    result
  end
  
  
end

end
